### PR TITLE
Major refactor of connection code

### PR DESCRIPTION
**Pull request by theJenix**
Connection code from HostActivity and FanActivity pulled into ConnectionService and MessagingService.

All connection management code now lives in ConnectionService.  This includes
bluetooth discovery, host/fan connection management, and sending and receiving
messages.

All messaging management code now lives in MessagingServices.  It receives new
IMessage objects from the ConnectionService, decomposes the messages into their
parts, and sends a broadcast intent/action to the rest of the system to
represent each message.  This decouples the message and network layer from the
rest of the app.

HostActivity and FanActivity have been modified to use these services.

Currently broken: remote fan discovery.  Not too far from being complete...

### CODE REVIEW

File: fanClub/src/com/lastcrusade/fanclub/FanActivity.java [hunk 1]
```diff
@@ -98,96 +72,87 @@ public void onClick(View v) {
...
-     */
-    protected void onAcceptedHost(BluetoothSocket socket) {
-        //disable discovery...we found our host.
-        BluetoothUtils.disableDiscovery(this);
-
-        //create the message thread for handling this connection
-        this.messageThread = new MessageThread(socket, this.messageDispatch);
-        this.messageThread.start();
-    }
+//    private void handleFoundFans(List<FoundFan> list) {
```
Comment (ejohnson44) [inline fanClub/src/com/lastcrusade/fanclub/FanActivity.java:137]: Why is all of this commented out?

Comment (theJenix) [inline fanClub/src/com/lastcrusade/fanclub/FanActivity.java:137]: that's the part thats not quite working yet (remote display of found fans).  If it's agreeable to everyone, my plan would be to pull this in, get it all integrated, and then revisit the remote finding of fans afterward.

File: fanClub/src/com/lastcrusade/fanclub/FanActivity.java [hunk 2]
```diff
@@ -98,96 +72,87 @@ public void onClick(View v) {
...
+        this.broadcastRegistrar
+            .addAction(MessagingService.ACTION_STRING_MESSAGE, new IBroadcastActionHandler() {
+
+                @Override
+                public void onReceiveAction(Context context, Intent intent) {
+                    String string = intent.getStringExtra(MessagingService.EXTRA_STRING);
+                    Toaster.iToast(FanActivity.this, string);
+                }
+            })
+//            .addAction(ConnectionService.ACTION_FIND_FINISHED, new IBroadcastActionHandler() {
```
Comment (bryansills) [inline fanClub/src/com/lastcrusade/fanclub/FanActivity.java:89]: why is this commented?

Comment (theJenix) [inline fanClub/src/com/lastcrusade/fanclub/FanActivity.java:89]: See above comment...this is part of the comment that isnt working yet.

**Pull request merged**